Library/Homebrew/cmd/install.rb

```diff
@@ -71,7 +71,13 @@ def install
 
         if f.installed?
           msg = "#{f.full_name}-#{f.installed_version} already installed"
-          msg << ", it's just not linked" unless f.linked_keg.symlink? || f.keg_only?
+          unless f.linked_keg.symlink?
+            if Pathname.new("#{HOMEBREW_CELLAR}/#{f.oldname}").exist?
+              msg << ", it's just not migrated"
```

Comment:
Revisit this PR, I think this is a dead code. The reason is if `f.installed?` returns true, the formula has to be migrated.